### ISSUE TITLE
JIRA - collector running, but no data

### ISSUE THREAD

**Issue by smsuresh** [capitalone/Hygieia#942]
Hi, 
 I just downloaded, brought up API & UI.
 bitbucket collector is partially up
 when I run jira collector, seeing the below in logs, which looks strange for me.
I never mentioned any ticket number, but its showing it..

----------------------

Below is the log
```
2016-11-14 08:11:45,179 INFO  o.s.j.e.a.AnnotationMBeanExporter - Registering beans for JMX exposure on startup
2016-11-14 08:11:45,193 INFO  com.capitalone.dashboard.Application - Started Application in 9.029 seconds (JVM running for 10.524)
2016-11-14 08:12:00,000 INFO  **c.c.d.collector.CollectorTask - Running Collector: Jira**
2016-11-14 08:12:00,008 INFO  c.c.d.collector.CollectorTask - -----------------------------------
2016-11-14 08:12:00,008 INFO  c.c.d.collector.CollectorTask - https://jira.mysite.com/
2016-11-14 08:12:00,008 INFO  c.c.d.collector.CollectorTask - -----------------------------------
2016-11-14 08:12:00,800 ERROR c.c.d.client.DefaultJiraClient - No result was available from Jira unexpectedly - defaulting to blank response. The reason for this fault is the following:RestClientException{statusCode=Optional.of(404), errorCollections=[]}
2016-11-14 08:12:00,801 INFO  c.c.d.collector.CollectorTask - Team Data               0        0s
2016-11-14 08:12:00,829 ERROR c.c.d.client.DefaultJiraClient - No result was available from Jira unexpectedly - defaulting to blank response. The reason for this fault is the following:RestClientException{statusCode=Optional.of(404), errorCollections=[]}
2016-11-14 08:12:00,829 INFO  c.c.d.collector.CollectorTask - Project Data            0        0s
2016-11-14 08:12:02,977 ERROR c.c.d.client.DefaultJiraClient - No result was available from Jira unexpectedly - defaulting to blank response. The reason for this fault is the following:RestClientException{statusCode=Optional.of(404), errorCollections=[]}
2016-11-14 08:12:02,978 INFO  c.c.d.collector.CollectorTask - Story Data              0        2s
2016-11-14 08:12:02,978 INFO  **c.c.d.collector.CollectorTask - Finished                         2s**
```
-----------------
BELOW IS MY property file
--------------------------
```properties
# PageSize - Expand contract this value depending on Jira implementation's
# default server timeout setting (You will likely receive a SocketTimeoutException)
feature.pageSize=100

# Delta change date that modulates the collector item task
# Occasionally, these values should be modified if database size is a concern
feature.deltaStartDate=2016-03-01T00:00:00.000000
feature.masterStartDate=2016-03-01T00:00:00.000000
feature.deltaCollectorItemStartDate=2016-03-01T00:00:00.000000

# Chron schedule: S M D M Y [Day of the Week]
feature.cron=0 * * * * *

# ST Query File Details - Required, but DO NOT MODIFY
feature.queryFolder=jiraapi-queries
feature.storyQuery=story
feature.epicQuery=epic

# JIRA CONNECTION DETAILS:
# Enterprise Proxy - ONLY INCLUDE IF YOU HAVE A PROXY
#feature.jiraProxyUrl=http://proxy.com
#feature.jiraProxyPort=9000
feature.jiraBaseUrl=https://jira.mysite.com/
feature.jiraQueryEndpoint=jira/rest/api/2/
# For basic authentication, requires username:password as string in base64
# This command will make this for you:  echo -n username:password | base64
feature.jiraCredentials=dXNoYW5zdTE6TX12312VydsdfWdhbjYh
# OAuth is not fully implemented; please blank-out the OAuth values:
feature.jiraOauthAuthtoken=
feature.jiraOauthRefreshtoken=
feature.jiraOauthRedirecturi=
feature.jiraOauthExpiretime=

#############################################################################
# In Jira, general IssueType IDs are associated to various "issue"
# attributes. However, there is one attribute which this collector's
# queries rely on that change between different instantiations of Jira.
# Please provide a string name reference to your instance's IssueType for
# the lowest level of Issues (e.g., "user story") specific to your Jira
# instance.  Note:  You can retrieve your instance's IssueType Name
# listings via the following URI:  https://[your-jira-domain-name]/rest/api/2/issuetype/
#############################################################################
feature.jiraIssueTypeId=Story

#############################################################################
# In Jira, your instance will have its own custom field created for "sprint" or "timebox" details,
# which includes a list of information.  This field allows you to specify that data field for your
# instance of Jira. Note: You can retrieve your instance's sprint data field name
# via the following URI, and look for a package name com.atlassian.greenhopper.service.sprint.Sprint;
# your custom field name describes the values in this field:
# https://[your-jira-domain-name]/rest/api/2/issue/[some-issue-name]
#############################################################################
feature.jiraSprintDataFieldName=customfield_10730

#############################################################################
# In Jira, your instance will have its own custom field created for "super story" or "epic" back-end ID,
# which includes a list of information.  This field allows you to specify that data field for your instance
# of Jira.  Note:  You can retrieve your instance's epic ID field name via the following URI where your
# queried user story issue has a super issue (e.g., epic) tied to it; your custom field name describes the
# epic value you expect to see, and is the only field that does this for a given issue:
# https://[your-jira-domain-name]/rest/api/2/issue/[some-issue-name]
#############################################################################
feature.jiraEpicIdFieldName=customfield_11330

#############################################################################
# In Jira, your instance will have its own custom field created for "story points"
# This field allows you to specify that data field for your instance
# of Jira.  Note:  You can retrieve your instance's storypoints ID field name via the following URI where your
# queried user story issue has story points set on it; your custom field name describes the
# story points value you expect to see:
# https://[your-jira-domain-name]/rest/api/2/issue/[some-issue-name]
#############################################################################
feature.jiraStoryPointsFieldName=customfield_10042
```

**Comment by smsuresh** [comment]
When I do the below:

>   and look for a package name com.atlassian.greenhopper.service.sprint.Sprint

im getting 0 result for all of my jira tickets, can you please let me know , how to populate the data for  

> **feature.jiraSprintDataFieldName**

**Comment by smsuresh** [comment]
I've verified all the **customfield_xxxxx,** they seem to be perfect for our jira.

**Comment by smsuresh** [comment]
@amp5208 @jimzucker  -- can you please help?!

**Comment by smsuresh** [comment]
![jira err](https://cloud.githubusercontent.com/assets/10332154/20259238/d3340668-aa79-11e6-9e30-e09a55207e34.png)

Above is the screen I see when I click "configure widget" for Feature (jira).

**Comment by MarkRx** [comment]
You are getting a 404 which means page not found. Your config must be wrong.

```
feature.jiraBaseUrl=https://jira.mysite.com/
feature.jiraQueryEndpoint=jira/rest/api/2/
```

My jira query endpoint is feature.jiraQueryEndpoint=rest/api/2/

**Comment by smsuresh** [comment]
https://jira.mysite.com/jira/rest/api/2/issue/TICKETID-5    -- Iam getting json response for this URL in browser,

but  getting below for  :   http://jira.mysite.com/jira/rest/api/2/


<status>
<status-code>404</status-code>
<message>
null for uri: http://jira.mysite.com/jira/rest/api/2/
</message>
</status>

----------
please suggest, how to proceed..

**Comment by smsuresh** [comment]
cool
Modified config as

feature.jiraBaseUrl=https://jira.mysite.com/jira/
feature.jiraQueryEndpoint=rest/api/2/

its working fine now!! :)

**Comment by smsuresh** [comment]
log is 


2016-11-15 05:45:10,287 INFO  com.capitalone.dashboard.Application - Started Application in 8.892 seconds (JVM running for 10.488)
2016-11-15 05:46:00,001 INFO  c.c.d.collector.CollectorTask - Running Collector: Jira
2016-11-15 05:46:00,008 INFO  c.c.d.collector.CollectorTask - -----------------------------------
2016-11-15 05:46:00,008 INFO  c.c.d.collector.CollectorTask - https://jira.mysite.com/jira/
2016-11-15 05:46:00,008 INFO  c.c.d.collector.CollectorTask - -----------------------------------
2016-11-15 05:46:06,370 INFO  c.c.d.collector.CollectorTask - Team Data             834        6s
2016-11-15 05:46:08,871 INFO  c.c.d.collector.CollectorTask - Project Data          834        2s


**But** still im seeing the same screen only..  not seeing any data..

![jira err](https://cloud.githubusercontent.com/assets/10332154/20294761/abbdcb52-ab26-11e6-8245-2d1d05d50c7a.png)

**Comment by smsuresh** [comment]
@MarkRx   can you please help?

**Comment by MarkRx** [comment]
Your API / UI / Jira collector must not be configured correctly. Ensure they are pointing to the same database, using the correct ports, and the data actually exists.

**Comment by smsuresh** [comment]
Thanks, JIRA was storing data in "dashboard" db where as all other components had the db as "dashboarddb"  -- BUT  why JIRA alone picking different DB when I configure for dashboarddb? can you pls calrify.

**Comment by smsuresh** [comment]
Can you clarify the purpose  of switches    ||      &  >>   under features..
It keep changing data, irrespective whether I pause or make it run.

**Comment by MarkRx** [comment]
The database defaults in different documents might be out of sync.

I suggest grabbing the latest code for jira. It allows configuration of the sprint type which might be what you want.

**Comment by nireeshT** [comment]
@smsuresh  - did @MarkRx  suggestion help you fix the issue.

**Comment by satishc1** [comment]
@smsuresh The feature widget now has an option to display metrics based on the number of tickets instead of estimates. This is for setups that do not use estimates.
Functionality merged in https://github.com/capitalone/Hygieia/pull/1054 Use latest codebase. You’ll also need to follow cleanup instructions provided in https://github.com/capitalone/Hygieia/pull/1137

**Comment by smsuresh** [comment]
Great!! Thankyou Sathish!!

Suresh.S

Sent from Yahoo Mail on Android 
 
  On Fri, Feb 3, 2017 at 9:25 PM, satishc1<notifications@github.com> wrote:   
Closed #942.

**Comment by RameshMuthuraj** [comment]
I'm getting the below response when i run the collector

2017-06-26 16:57:00,014 INFO  c.c.d.collector.CollectorTask - Team Data               0        0s
2017-06-26 16:57:00,349 ERROR c.c.d.client.DefaultJiraClient - No result was available from Jira unexpectedly - defaulting to blank response. The reason for this fault is the following:RestClientException{statusCode=Optional.of(404), errorCollections=[]}
2017-06-26 16:57:00,349 INFO  c.c.d.collector.CollectorTask - Project Data            0        0s
2017-06-26 16:57:00,715 ERROR c.c.d.client.DefaultJiraClient - Parser exception when parsing statuses
org.json.simple.parser.ParseException: null
        at org.json.simple.parser.Yylex.yylex(Yylex.java:610) ~[json-simple-1.1.1.jar!/:na]
        at org.json.simple.parser.JSONParser.nextToken(JSONParser.java:269) ~[json-simple-1.1.1.jar!/:na]
        at org.json.simple.parser.JSONParser.parse(JSONParser.java:118) ~[json-simple-1.1.1.jar!/:na]
        at org.json.simple.parser.JSONParser.parse(JSONParser.java:81) ~[json-simple-1.1.1.jar!/:na]
        at org.json.simple.parser.JSONParser.parse(JSONParser.java:75) ~[json-simple-1.1.1.jar!/:na]
        at com.capitalone.dashboard.client.DefaultJiraClient.getStatusMapping(DefaultJiraClient.java:365) ~[jira-feature-collector.jar!/:2.0.5-SNAPSHOT]


Please help me to fix this 

#Jira Connection Details
#feature.jiraProxyUrl=
#feature.jiraProxyPort=8081
feature.jiraBaseUrl=https://jira.tools/login.jsp?
feature.jiraCredentials=XXXXXX
feature.jiraQueryEndpoint=rest/api/2/
feature.jiraOauthAuthtoken=
feature.jiraOauthRefreshtoken=
feature.jiraOauthRedirecturi=
feature.jiraOauthExpiretime=
